source/dlangbot/ci.d

```diff
@@ -0,0 +1,169 @@
+module dlangbot.ci;
+
+import dlangbot.github;
+
+import vibe.core.log;
+import vibe.http.client : HTTPClientRequest, requestHTTP;
+
+import std.conv : to;
+import std.format : format;
+import std.regex : matchFirst, regex;
+import std.exception;
+import std.variant : Nullable;
+
+// list of used APIs (overwritten by the test suite)
+string dTestAPI = "http://dtest.dlang.io";
+string circleCiAPI = "https://circleci.com/api/v1.1";
+string projectTesterAPI = "https://ci.dawg.eu";
+
+// only since 2.073
+auto nullable(T)(T t) {  return Nullable!T(t); }
+
+/**
+There's no way to get the PR number from a GitHub status event (or other API
+endpoints).
+Hence we have to check the sender for this information.
+*/
+Nullable!uint getPRForStatus(string repoSlug, string url, string context)
+{
+    Nullable!uint prNumber;
+
+    try {
+        logDebug("getPRNumber (repo: %s, ci: %s)", repoSlug, context);
+        switch (context) {
+            case "auto-tester":
+                prNumber = checkAutoTester(url);
+                break;
+            case "ci/circleci":
+                prNumber = checkCircleCi(url);
+                break;
+            case "continuous-integration/travis-ci/pr":
+                prNumber = checkTravisCi(url);
+                break;
+            case "CyberShadow/DAutoTest":
+                prNumber = checkDTest(url);
+                break;
+            case "Project Tester":
```

Comment:
Please add the new Jenkins, should be `continuous-integration/jenkins/pr-merge`.